pkg/controller/common/helpers.go

```diff
@@ -558,42 +559,27 @@ func removeIgnDuplicateFilesUnitsUsers(ignConfig ign2types.Config) (ign2types.Co
 // internally it transpiles to Ign spec v3 config
 func TranspileCoreOSConfigToIgn(files, units []string) (*ign3types.Config, error) {
 	overwrite := true
-	outConfig := ign3types.Config{}
+	buCfg := bubase.Config{}
 	// Convert data to Ignition resources
 	for _, d := range files {
-		f := new(fcctbase.File)
+		f := new(bubase.File)
 		if err := yaml.Unmarshal([]byte(d), f); err != nil {
 			return nil, fmt.Errorf("failed to unmarshal file into struct: %v", err)
 		}
 		f.Overwrite = &overwrite
 
-		// Add the file to the config
-		var ctCfg fcctbase.Config
-		ctCfg.Storage.Files = append(ctCfg.Storage.Files, *f)
-		ign3_0config, tSet, err := ctCfg.ToIgn3_0()
-		if err != nil {
-			return nil, fmt.Errorf("failed to transpile config to Ignition config %s\nTranslation set: %v", err, tSet)
-		}
-		ign3_2config := translate3.Translate(translate3_1.Translate(ign3_0config))
-		outConfig = ign3.Merge(outConfig, ign3_2config)
+		buCfg.Storage.Files = append(buCfg.Storage.Files, *f)
```

Comment:
While I had the patient open, I changed this to generate a single butane config, then translate it once to ignition which is way simpler than generating a tiny butane config and then continually merging ignition.